src/xalloc.c

```diff
@@ -138,7 +358,7 @@ void *xmalloc(size_t size)
         return NULL;                          // during cleanup we must not jump again
     }
 
-    insert_ptr(p);
+    insert_ptr_legacy(p);

```

Comment:
You can replace the guts of this function with just 'return arena_xmalloc(&global_arena, size);  No need to duplicate work that occurs in the arena_xmalloc function.